eth/stagedsync/sync.go

```diff
@@ -137,25 +135,6 @@ func (s *Sync) IsAfter(stage1, stage2 stages.SyncStage) bool {
 
 func (s *Sync) HasUnwindPoint() bool { return s.unwindPoint != nil }
 func (s *Sync) UnwindTo(unwindPoint uint64, reason UnwindReason, tx kv.Tx) error {
-	if tx != nil {
-		if casted, ok := tx.(state.HasAggTx); ok {
-			// protect from too far unwind
-			unwindPointWithCommitment, ok, err := casted.AggTx().(*state.AggregatorRoTx).CanUnwindBeforeBlockNum(unwindPoint, tx)
-			// Ignore in the case that snapshots are ahead of commitment, it will be resolved later.
-			// This can be a problem if snapshots include a wrong chain so it is ok to ignore it.
-			if errors.Is(err, state.ErrBehindCommitment) {
-				return nil
-			}
-			if err != nil {
-				return err
-			}
-			if !ok {
-				return fmt.Errorf("too far unwind. requested=%d, minAllowed=%d", unwindPoint, unwindPointWithCommitment)
-			}
-			unwindPoint = unwindPointWithCommitment
```

Comment:
but we need replace it by "unwind only to blocks where we have diffs". 
and likely gaps in diffs are not allowed?

Reply:
I can do in a follow-up where I remove old unwinds, it probably makes more sense there

Reply:
As I remember `CanUnwindBeforeBlockNum` has a lot of features: it doesn't allows unwind to 0, etc...